source/plugins/go/src/network_flow_logs.go

```diff
@@ -0,0 +1,380 @@
+				if MdsdNetworkFlowLogsStreamTagName == "" {
+					Log("Error::mdsd::Failed to get stream tag for networkflow logs. Will retry ...")
+					return output.FLB_RETRY
+				}
+			}
+			if MdsdNetworkFlowClient == nil {
+				Log("Error::mdsd::mdsd connection does not exist for networkflow mdsd client. re-connecting ...")
+				CreateMDSDClient(NetworkFlowLogs, ContainerType)
+				if MdsdNetworkFlowClient == nil {
+					Log("Error::mdsd::Unable to create mdsd client for networkflow. Please check error log.")
+					NetworkFlowTelemetryMutex.Lock()
+					defer NetworkFlowTelemetryMutex.Unlock()
+					NetworkFlowLogsMDSDClientCreateErrors += 1
+
+					return output.FLB_RETRY
+				}
+			}
+
+			bts, er := writeNetworkFlowMsgPackEntries(MdsdNetworkFlowClient, MdsdNetworkFlowLogsStreamTagName, networkFlowLogsMsgPackEntries)
+			elapsed = time.Since(start)
+
+			if er != nil {
+				Log("Error::mdsd::Failed to write to mdsd %d records for networkflow logs after %s. Will retry ... error : %s", len(networkFlowLogsMsgPackEntries), elapsed, er.Error())
+				if MdsdNetworkFlowClient != nil {
+					MdsdNetworkFlowClient.Close()
+					MdsdNetworkFlowClient = nil
+				}
+
+				NetworkFlowTelemetryMutex.Lock()
+				defer NetworkFlowTelemetryMutex.Unlock()
+				NetworkFlowLogsMDSDClientCreateErrors += 1
+
+				return output.FLB_RETRY
+			} else {
+				numNetworkLogRecords = len(networkFlowLogsMsgPackEntries)
+				Log("Success::mdsd::Successfully flushed %d networkflow log records that was %d bytes to mdsd in %s ", numNetworkLogRecords, bts, elapsed)
+			}
+		}
+
+		//TODO Telemetry
+		// NetworkFlowTelemetryMutex.Lock()
+		// defer NetworkFlowTelemetryMutex.Unlock()
+
+		// if numNetworkLogRecords > 0 {
+		// 	FlushedRecordsCount += float64(numNetworkLogRecords)
+		// 	FlushedRecordsTimeTaken += float64(elapsed / time.Millisecond)
+
+		// 	if maxLatency >= AgentLogProcessingMaxLatencyMs {
+		// 		AgentLogProcessingMaxLatencyMs = maxLatency
+		// 	}
+		// }
+	}
+	return output.FLB_OK
+}
+
+func convertFluentBitRecord(input interface{}) (interface{}, error) {
+	switch v := input.(type) {
+	case map[interface{}]interface{}:
+		record := make(map[string]interface{})
+		for key, value := range v {
+			strKey, ok := key.(string)
+			if !ok {
+				strKey = fmt.Sprintf("%v", key)
+			}
+			convertedValue, err := convertFluentBitRecord(value)
+			if err != nil {
+				return nil, err
+			}
+			record[strKey] = convertedValue
+		}
+		return record, nil
+	case []byte:
+		return string(v), nil
+	case []interface{}:
+		records := make([]interface{}, len(v))
+		for i, item := range v {
+			convertedItem, err := convertFluentBitRecord(item)
+			if err != nil {
+				return nil, err
+			}
+			records[i] = convertedItem
+		}
+		return records, nil
+	default:
+		return v, nil
+	}
+}
+
+func mapNetworkFlowLogsToDataMap(dataMap map[string]interface{}, record map[string]interface{}) error {
```

Comment:
what's the max Network flow logs message size tested? we should add telemetry for NetworkFlowLogRecord message size and see if we are hitting 64KB size limit in the ingestion pipeline.